keras/src/backend/common/variables.py

```diff
@@ -211,6 +211,12 @@ def __init__(
 
     def _deferred_initialize(self):
         if self._value is not None:
+            # If NNX is enabled, it's possible the variable was already
+            # initialized by a concrete call. In this case,
+            # _deferred_initialize becomes a no-op for this variable.
+            if config.is_nnx_enabled():
+                self._initializer = None  # Clear initializer as it's now "used"
```

Comment:
This NNX specific code path looks suspicious, I think it's hiding a bug.

The only way that this could happen (from what I can see) is that the value generated in a stateless scope is captured. I'm surprised this can work because it should be a TracedArray instead of a concrete array and should fail later when used in a non-tracing context.

Do you have an example of where it fails without this code?